pkg/ovs/volumes.go

```diff
@@ -0,0 +1,215 @@
+package ovs
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GetVolumes -
+func GetVolumes(name string) []corev1.Volume {
+
+	var scriptsVolumeDefaultMode int32 = 0755
+	directoryOrCreate := corev1.HostPathDirectoryOrCreate
+
+	//source_type := corev1.HostPathDirectoryOrCreate
+	return []corev1.Volume{
+		{
+			Name: "etc-machine-id",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/machine-id",
+				},
+			},
+		},
+		{
+			Name: "etc-localtime",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/etc/localtime",
+				},
+			},
+		},
+		{
+			Name: "etc-ovs",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/etc/ovs",
+					Type: &directoryOrCreate,
+				},
+			},
+		},
+		{
+			Name: "var-run",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/var/run/openvswitch",
+					Type: &directoryOrCreate,
+				},
+			},
+		},
+		{
+			Name: "var-log",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/var/log/openvswitch",
+					Type: &directoryOrCreate,
+				},
+			},
+		},
+		{
+			Name: "var-lib",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/var/lib/openvswitch",
+					Type: &directoryOrCreate,
+				},
+			},
+		},
+		{
+			Name: "var-run-ovn",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/var/run/ovn",
+					Type: &directoryOrCreate,
+				},
+			},
+		},
+		{
+			Name: "var-log-ovn",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/var/log/ovn",
+					Type: &directoryOrCreate,
+				},
+			},
+		},
+		{
+			Name: "var-lib-ovn",
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: "/var/home/core/openstack/var/lib/ovn",
```

Comment:
what are the requirements for those HostPaths? persist for the lifetime of the pod, or just persist forever?

Reply:
actually that one isn't needed at all. I removed it.
For others, I want to have persistent ovsdb stored on host so it will e.g. survive host reboot.